### PR TITLE
v0.3.1-alpha.1: Stabilization + Oracle B-O1/B-O2 + Full Medium/High completion

### PR DESCRIPTION
**Pull request by InitusNovus**
## Summary

This PR merges 80 commits from `dev` into `main`, representing the stabilization pause checkpoint for version `0.3.1-alpha.1`. It completes all Critical, High, Medium, and select Low ROADMAP items, plus two Oracle hardening milestones (B-O1, B-O2).

**Scope**: This is an internal alpha pre-release. The actual `v0.3.1` release tag will be applied to `main` after PR review and merge.

---

### 🔮 Oracle Hardening (B-O1, B-O2)
- **B-O1**: Full Oracle corpus integration test suite — 91,623 total test cases, 99.25% adjusted pass rate, 0 skipped
- **B-O2**: Mux per-branch vector generation in `engine.py`; added mux-specific unit tests; catalogued 1,778 Category-C exceptions in `CATEGORY_C_EXCEPTIONS.md`
- Residual tech debt cleaned: removed unused `db` parameter from `_generate_mux_vectors()`; added B-O3 TODO

### 🏗️ Core F# Library
- All Critical (C-1, C-2) and High (H-1, H-2, H-3) items complete
- All Medium (M-1 through M-4) items complete
- Low item L-4 complete

### 📦 Version Bump
- `Signal.CANdy.Core.fsproj`, `Signal.CANdy.fsproj`, `Api.fs` all updated to `0.3.1-alpha.1`
- Public-facing README install examples remain at `0.3.0` (alpha not exposed to end users)

### 📚 Documentation & History
- Comprehensive session reports migrated into `Reports/` (append-only historical record)
- ROADMAP fully updated with all completed checkboxes
- `ORACLE_RESULTS.md` and `CATEGORY_C_EXCEPTIONS.md` maintained as immutable evidence

---

## Oracle Corpus Results

| Metric | Value |
|--------|-------|
| Total test cases | 91,623 |
| Adjusted pass rate | 99.25% |
| Category-C failures | 1,778 |
| Skipped | 0 |

Category-C failures are documented exceptions (edge cases in DBC files, not generator bugs) — see `tests/oracle/CATEGORY_C_EXCEPTIONS.md`.

---

### dotnet test (Release)
- Signal.CANdy.Core.Tests: **73 passed**, 0 failed ✅
- Generator.Tests: **27 passed**, 0 failed ✅
- Total: **100 passed**, 0 failed ✅

### Python pytest (Oracle — local only, not in CI)
- **41 passed**, 0 failed ✅

### dotnet build
- **0 warnings, 0 errors** ✅

### CI (GitHub Actions — build-test job)
- All steps passed: F# tests, codegen, C build, C++ compat, smoke test, NuGet pack ✅
- Note: `lint` job fails on Fantomas — this is a **pre-existing** issue in `Codegen.fs`, `Dbc.fs`, `CodegenTests.fs`, `DbcTests.fs` that predates this PR. Our changed file (`Api.fs`) passes Fantomas cleanly.

---

## Version Note

- This PR introduces `0.3.1-alpha.1` — an **internal alpha** for validation before the actual `v0.3.1` release
- After merge: apply `v0.3.1` tag on `main` for the real NuGet release
- NuGet README install examples will be updated at that point

---

## Remaining Backlog (out of scope for this PR)

- **B-O3**: Pure extended-mux message support (very rare in practice; tracked via TODO comment)
- **L-1, L-2, L-3**: Low-priority feature improvements
- **Fantomas**: Fix pre-existing formatting in `Codegen.fs`, `Dbc.fs`, `CodegenTests.fs`, `DbcTests.fs`

---

## Commit Count

**80 commits** ahead of `main`. See `git log --oneline origin/main..HEAD` for full history.

### CODE REVIEW

File: tests/oracle/ORACLE_RESULTS.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+1#JY|# Oracle Test Pipeline - Integration Results
+2#KM|
+3#ZQ|**Date**: 2026-03-12 (B-O2 update)
```
[the remark this answers is not in the document]
Comment (InitusNovus) [reply tests/oracle/ORACLE_RESULTS.md]: Confirmed and fixed. The file had accidental per-line prefixes. Commit dda53aa cleans tests/oracle/ORACLE_RESULTS.md to plain Markdown. Verified with regex that no prefixed marker lines remain under tests/oracle/*.md.
